### PR TITLE
[MLc] fix computation for distanceMode = 'hypocentral'

### PR DESCRIPTION
**Pull request by luca-s**
The current implementation considers the origin depth as the vertical distance to the receiver, which is an assumption that breaks at local scale. The fix computes the vertical distance as the difference between sensor elevation and origin depth.

### CODE REVIEW

File: libs/seiscomp/processing/magnitudes/MLc.cpp [hunk 1]
```diff
@@ -264,12 +264,20 @@ MagnitudeProcessor::Status MagnitudeProcessor_MLc::computeMagnitude(
...
 
 	SEISCOMP_DEBUG("  + distance mode: %s", distanceMode.c_str());
+
+	double hDistanceKm = Math::Geo::deg2km(delta);
+	double vDistanceKm = 0;
+	if ( !receiver && distanceMode == "hypocentral" ) {
+		return MetaDataRequired;
+	}
+	if ( receiver ) {
+		vDistanceKm = abs(receiver->elevation() / 1000. + depth);
```
Comment (gempa-dirk) [inline libs/seiscomp/processing/magnitudes/MLc.cpp:274]: maybe abs() is not required here and it even would hide the information if the event is below or  above the sensor  which is also reported in debug output.

**Comment by gempa-dirk** [comment]
The changes are reasonable.

File: libs/seiscomp/processing/magnitudes/MLc.cpp (hunk 1 above)
Comment (luca-s) [reply libs/seiscomp/processing/magnitudes/MLc.cpp:274]: Good point.

File: libs/seiscomp/processing/magnitudes/MLc.cpp [hunk 2]
```diff
@@ -264,12 +264,20 @@ MagnitudeProcessor::Status MagnitudeProcessor_MLc::computeMagnitude(
...
+		vDistanceKm = receiver->elevation() / 1000. + depth;
+	}
 	double distanceKm;
 	if ( distanceMode == "hypocentral" ) {
-		distanceKm = sqrt(pow(Math::Geo::deg2km(delta),2) + pow(depth,2));
-	}
-	else {
-		distanceKm = Math::Geo::deg2km(delta);
+		distanceKm = sqrt(pow(hDistanceKm, 2) + pow(vDistanceKm,2));
+	} else {
```
Comment (gempa-jabe) [inline libs/seiscomp/processing/magnitudes/MLc.cpp:279]: FYI: we want a newline before `else`. I am going to fix that after merging.

**Pull request merged**